R/drop_download.R

```diff
@@ -0,0 +1,128 @@
+#' Download a file from Dropbox to disk.
+#'
+#' @param path path to a file in Dropbox
+#' @param local_path path to save file to. If NULL (the default), saves file to working directory with same name. If not null, but a valid folder, file will be saved in this folder with same basename as path. If not null and not a folder, file will be saved to this path exactly.
+#' @param overwrite If TRUE, overwrite local file. Defaults to FALSE
+#' @param progress If TRUE, show a progress bar for large file downloads. Defaults to TRUE in interactive sessions, otherwise FALSE.
+#' @param verbose if TRUE, emit message giving location and size of the newly downloaded file. Defaults to TRUE in interactive sessions, otherwise FALSE.
+#' @template token
+#'
+#' @return path to location file downloaded to, invisibly
+#'
+#' @examples \dontrun{
+#'
+#'   # download a file to the current working directory
+#'   drop_get("dataset.zip")
+#'
+#'   # download again, overwriting previous result
+#'   drop_get("dataset.zip", overwrite = TRUE)
+#'
+#'   # download to a different path, keeping file name constant
+#'   # will download to "some/other/place/dataset.zip"
+#'   drop_get("dataset.zip", local_file = "some/other/place/")
+#'
+#'   # download to to a different path, changing filename
+#'   drop_get("dataset.zip", local_file = "some/other/place/not_a_dataset.zip")
+#' }
+#'
+#' @export
+drop_download <- function(
+  path,
+  local_path = NULL,
+  overwrite = FALSE,
+  progress = interactive(),
```

Comment:
How do the progress bars work? I could not get them to even for large(ish) files. But this is somewhat cosmetic.

Reply:
yeah, same; I left this in because it was in the old version, and I trusted `httr` to "to it right"; should I just remove those args?